src/material-experimental/mdc-dialog/_mdc-dialog-structure-overrides.scss

```diff
@@ -0,0 +1,42 @@
+// Mixin that can be included to override the default MDC dialog styles to fit
+// our needs. See individual comments for context on why certain styles need to be modified.
+@mixin _mdc-dialog-structure-overrides() {
+  // MDC dialog sets max-height and max-width on the `mdc-dialog__surface` element. This
+  // element is the parent of the portal outlet. This means that the actual user-content
+  // is scrollable, but as per Material Design specification, only the dialog content
+  // (indicated by `matDialogContent`) should be scrollable while title and actions are fixed.
+  // This is not an issue for MDC because they make the assumption that the content, title and
+  // action elements are direct children of the surface element. We work around this by setting
+  // an explicit max-height for the content element, so that the content is scrollable. This
+  // matches the behavior from the non-MDC dialog and is backwards compatible.
+  .mat-mdc-dialog-content {
+    max-height: $mat-dialog-content-max-height;
+  }
+
+  .mat-mdc-dialog-container {
+    // MDC by default expands the dialog using a fixed position to the viewport dimensions.
+    // MDC does this because the dialog element contains the backdrop/scrim too. This is not
+    // the case for our implementation of the dialog that uses the CDK overlay backdrop. We
+    // update the dialog to a `static` position so that the dialog scales based on the dimensions
+    // passed to the CDK overlay container (allowing for custom position strategies).
+    // Additionally, the MDC dialog is designed to always stay in the DOM. Hence MDC sets
+    // `display: none` when the dialog is closed. Since we only attach the dialog to the DOM
+    // when open, and we want to be able to focus the dialog immediately when we attach it, we
+    // set override the default dialog so that the dialog is focusable when attached.
+    &.mdc-dialog {
+      position: static;
+      display: flex;
```

Comment:
We aren't supposed to use `flex` around the user's content. Can we get away with `block` or potentially add another wrapper `div` around it?

Reply:
MDC heavily relies on `flex` in their dialog. I added a small override for this w/ comments. I also saw a compatibility issue related to this, so it definitely makes sense to do this (even if it is just for the interim)